front/components/poke/subscriptions/table.tsx

```diff
@@ -59,180 +72,329 @@ export function SubscriptionsDataTable({
+}: {
+  show: boolean;
+  onClose: () => void;
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+}) {
+  const router = useRouter();
+  const { plans } = usePokePlans();
+
+  const { submit: onDowngrade } = useSubmitFunction(async () => {
+    if (
+      !window.confirm(
+        "Are you sure you want to downgrade this workspace to no plan?"
+      )
+    ) {
+      return;
+    }
+    try {
+      const r = await fetch(`/api/poke/workspaces/${owner.sId}/downgrade`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      if (!r.ok) {
+        throw new Error("Failed to downgrade workspace.");
+      }
+      router.reload();
+    } catch (e) {
+      console.error(e);
+      window.alert("An error occurred while downgrading the workspace.");
+    }
+  });
+
+  const { submit: onUpgradeToPlan } = useSubmitFunction(
+    async (plan: PlanType) => {
+      if (
+        !window.confirm(
+          `Are you sure you want to upgrade ${owner.name} (${owner.sId}) to plan ${plan.name} (${plan.code}) ?.`
+        )
+      ) {
+        return;
+      }
+      try {
+        const r = await fetch(
+          `/api/poke/workspaces/${owner.sId}/upgrade?planCode=${plan.code}`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+        if (!r.ok) {
+          throw new Error("Failed to upgrade workspace to plan.");
+        }
+        router.reload();
+      } catch (e) {
+        console.error(e);
+        window.alert(
+          "An error occurred while upgrading the workspace to plan."
+        );
+      }
+    }
+  );
+
+  return (
+    <Modal
+      isOpen={show}
+      onClose={onClose}
+      hasChanged={false}
+      title="Upgrade / Downgrade Workspace"
+      variant="full-screen"
+    >
+      <Page>
+        <Page.SectionHeader
+          title="Downgrade Workspace"
+          description="This action will downgrade the workspace to having no plan. This means that all the features will be disabled and members of
+          the workspaces will be redirected to the paywall page. After 15 days, the workspace data will be deleted."
+        />
+        <div>
+          <Button
+            label="Downgrade to NO PLAN"
+            variant="secondaryWarning"
+            onClick={onDowngrade}
+            disabled={subscription.plan.code === FREE_NO_PLAN_CODE}
+          />
+        </div>
+        <Separator />
+        <Page.SectionHeader
+          title="Upgrade Workspace to a Free Plan"
+          description="This action will upgrade the workspace to a free plan. This means that all the features will be enabled and members of the workspace will be able to use the workspace according to the selected plan product limitations."
+        />
+        <div>
+          {plans
+            // Daph Uncomment this line when Enteprise billing Form is ready
```

Comment:
Here to filter on FREE plan instead of relying on plan.billingType.
For now I'm just excluding PRO plan, and once we have the Enterprise plan form (my current task) I can uncomment and display only free plans here.